include/IECore/MeshAlgo.h

```diff
@@ -54,6 +54,8 @@ std::pair<PrimitiveVariable, PrimitiveVariable> calculateTangents( const MeshPri
 	const std::string &position = "P"
 );
 
+PrimitiveVariable resamplePrimitiveVariable( const MeshPrimitive *mesh, const std::string &primitiveVariable, PrimitiveVariable::Interpolation interpolation );
```

Comment:
I might be being dumb, but I don't know what the "along with const PrimitiveVariable&" bit is referring to - 4) would need a non-const reference.

I wouldn't mind a bit of time to think about the pointer vs reference thing - references do communicate the non-null requirement, but they can make it more painful to do things like choose which mesh to work on beforehand (because assignment is value assignment). Have you got any experience with that new gsl::not_null thingy? Seems like that maybe keeps the flexibility of flinging pointers around, but is a stronger guard against nulls...